tests/network/primary_pod_network.go

```diff
@@ -70,25 +70,66 @@ var _ = SIGDescribe("[Serial]Primary Pod Network", func() {
 		})
 
 		Context("VMI connected to the pod network using bridge binding", func() {
-			var vmi *v1.VirtualMachineInstance
+			When("Guest Agent exists", func() {
+				var (
+					vmi   *v1.VirtualMachineInstance
+					vmiIP = func() string {
+						var err error
+						vmi, err = virtClient.VirtualMachineInstance(vmi.Namespace).Get(vmi.Name, &metav1.GetOptions{})
```

Comment:
Question, lets say this Get doesnt success,
we saw on #4472 that a Get which doesnt success, you cant trust the return value
(i dont know by heart to say what the return value would exactly have it in fields, but it wont be nil)
Then `AfterEach` would silently fail on the Delete, because vmi != nil and the Delete has Expect.

We might dont care for it, but raising it for your consideration.